games/achievements.py

```diff
@@ -42,101 +52,193 @@ class DNFAchievement(Achievement):
+        elif (
             Game.add_durations(
                 Game.objects.filter(
-                    gameplayer__in=user.gameplayer_set.filter(dnf=False).filter(
-                        game__dnf=False
+                    gameplayer__in=user.gameplayer_set.filter(
+                        dnf=False, game__dnf=False
                     )
                 )
             )
             .filter(duration__lt=datetime.timedelta(minutes=30))
             .exists()
-        )
+        ):
+            return AchievementLevel.BASE
+        else:
+            return AchievementLevel.NO_LEVEL
 
 
 class DanishDSTAchievement(Achievement):
     name = "DST"
     description = "Participated in a game, while a DST transition happened in Denmark"
     icon = "backward-time.svg"
 
-    def has_achieved(user):
+    def get_level(user):
         query = Q()
         for dt in DST_TRANSITION_TIMES:
             query |= Q(start_datetime__lt=dt, end_datetime__gt=dt)
 
-        return user.games.filter(query).exists()
+        if user.games.filter(query).exists():
+            return AchievementLevel.BASE
+        return AchievementLevel.NO_LEVEL
 
 
 class TheBarrelAchievement(Achievement):
     name = "The Barrel"
-    description = "Consumed 100 beers in-game"
+    description = "Consumed (100/1000/2000/4000) beers in-game"
     icon = "barrel.svg"
 
-    def has_achieved(user):
-        return (user.stats_for_season(all_time_season).total_sips / 14) >= 100
+    def get_level(user):
+        total_sips = user.stats_for_season(all_time_season).total_sips / 14
+        if total_sips >= 4000:
+            return AchievementLevel.GOLD
+        elif total_sips >= 2000:
+            return AchievementLevel.SILVER
+        elif total_sips >= 1000:
+            return AchievementLevel.BRONZE
+        elif total_sips >= 100:
+            return AchievementLevel.BASE
+        return AchievementLevel.NO_LEVEL
 
 
 class BundeCampAchievement(Achievement):
     name = "Chug Camp"
-    description = "Got 50 chugs in-game"
+    description = "Got (50/100/250/500) chugs in-game"
     icon = "ace.svg"
 
-    def has_achieved(user):
-        return user.stats_for_season(all_time_season).total_chugs >= 50
+    def get_level(user):
+        total_chugs = user.stats_for_season(all_time_season).total_chugs
+        if total_chugs >= 500:
+            return AchievementLevel.GOLD
+        elif total_chugs >= 250:
+            return AchievementLevel.SILVER
+        elif total_chugs >= 100:
+            return AchievementLevel.BRONZE
+        elif total_chugs >= 50:
+            return AchievementLevel.BASE
+        return AchievementLevel.NO_LEVEL
 
 
 class StudyHardAchievement(Achievement):
     name = "Study Hard"
-    description = (
-        "Spend at least the amount of time corresponding to 2.5 ECTS in game (56 hours)"
-    )
+    description = "Spend at least the amount of time corresponding to (2.5/Quarter/Semester/Year) ECTS in game"
```

Comment:
I think this is more clear:

```suggestion
    description = "Spend at least the amount of time corresponding to (2.5/15/30/60) ECTS in game"
```

Reply:
Words for "Humorous effect", changed for clarity(RIP Quarter system)